hoot-services/src/main/java/hoot/services/controllers/job/DeriveChangesetResource.java

```diff
@@ -0,0 +1,164 @@
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2016 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+package hoot.services.controllers.job;
+
+import static hoot.services.HootProperties.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.xml.transform.dom.DOMSource;
+
+import org.apache.commons.io.FileUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.w3c.dom.Document;
+
+import hoot.services.models.db.JobStatus;
+import hoot.services.utils.XmlDocumentBuilder;
+
+
+/**
+ * Derives an OSM XML changeset (.osc)
+ * 
+ * Keeping this logic separate from that of ChangesetResource since it is not part of the official 
+ * OSM API and ChangesetResource is made up of pure OSM API functions.
+ */
+@Controller
+@Path("/changeset")
+public class DeriveChangesetResource extends JobControllerBase {
+    @SuppressWarnings("unused")
+    private static final Logger logger = LoggerFactory.getLogger(DeriveChangesetResource.class);
+
+    public DeriveChangesetResource() {
+        super(DERIVE_CHANGESET_SCRIPT);
+    }
+
+    /**
+     * Derives the difference between two OSM datasets, where the first dataset is a source and the 
+     * other a target.  The difference is derived in the form of an OSM XML changeset.
+     * 
+     * @param input1 database layer name of the first input dataset
+     * @param input2 database layer name of the second input dataset
+     * @param aoi bounding box used to filter the changeset derivation
+     * @return job ID of the changeset derivation job
+     */
+    @POST
+    @Path("/derive")
+    @Produces(MediaType.APPLICATION_JSON)
+    public JobId derive(@QueryParam("input1") String input1, @QueryParam("input2") String input2, @QueryParam("aoi") String aoi) {
+        
+        String jobId = UUID.randomUUID().toString();
+        jobId = "ex_" + jobId.replace("-", "");
+
+        try {
+            JSONArray commandArgs = new JSONArray();
+            JSONObject arg = new JSONObject();
+            
+            //assuming that this would only be used with data layers already being loaded into the
+            //database
+            arg.put("input1", DB_URL + "/" + input1);
```

Comment:
You're right, looks like I confused myself and botched this according to the workflow we've decided on.  DeriveChangesetResource::derive should look like: 

public JobId derive(@QueryParam("sourceLayer") String sourceLayer, @QueryParam("aoi") String aoi)

where sourceLayer is the hoot layer and then the target layer will be whatever OSM API database hoot is configured with.  I'll make those changes.